python_modules/dagster/dagster/_core/definitions/parent_asset_graph_differ.py

```diff
@@ -0,0 +1,122 @@
+from enum import Enum
+from typing import TYPE_CHECKING, Callable, Optional, Sequence, Union
+
+import dagster._check as check
+from dagster._core.definitions.external_asset_graph import ExternalAssetGraph
+from dagster._core.workspace.context import BaseWorkspaceRequestContext
+
+if TYPE_CHECKING:
+    from dagster._core.definitions.asset_graph import AssetGraph
+    from dagster._core.definitions.events import (
+        AssetKey,
+    )
+
+
+class ChangeReason(Enum):
+    NEW = "NEW"
+    CODE_VERSION = "CODE_VERSION"
+    INPUTS = "INPUTS"
+
+
+class ParentAssetGraphDiffer:
+    """Given two asset graphs, parent_asset_graph and branch_asset_graph, we can compute how the
+    assets in branch_asset_graph have changed with respect to parent_asset_graph. The ChangeReason
+    enum contains the list of potential changes an asset can undergo.
+    """
```

Comment:
nit:
hmm, wondering if we should do `branch` & `base` or `parent` & `child` or `old` & `new`

`parent` and `branch` is probably ok but feels a little mixed up

Reply:
i like branch and base